PipelineQueryWithinSameForeverTest.testPipelineQuery failing intermittently

**Description:**
This test fails intermittently.

```
[ERROR] Tests run: 143, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 283.165 s <<< FAILURE! - in TestSuite
[ERROR] testPipelineQuery(org.ballerinalang.stdlib.streams.legacy.PipelineQueryWithinSameForeverTest)  Time elapsed: 2.136 s  <<< FAILURE!
java.lang.AssertionError: Expected events are not received expected [1] but found [0]
	at org.ballerinalang.stdlib.streams.legacy.PipelineQueryWithinSameForeverTest.testPipelineQuery(PipelineQueryWithinSameForeverTest.java:49)
```
Complete log - https://api.travis-ci.org/v3/job/485746348/log.txt

--- a/stdlib/streams/src/test/resources/test-src/legacy/pipeline-streaming-test-v2.bal
+++ b/stdlib/streams/src/test/resources/test-src/legacy/pipeline-streaming-test-v2.bal
@@ -72,7 +72,14 @@ function startPipelineQuery() returns (StatusCount[]) {
     teacherStream6.publish(t2);
     teacherStream6.publish(t3);
 
-    runtime:sleep(1000);
+    int count = 0;
+    while(true) {
+        runtime:sleep(500);
+        count += 1;
+        if((globalStatusCountArray.length()) == 1 || count == 10) {
+            break;
+        }
+    }
 
     return globalStatusCountArray;
 }

